.doom.d/config.org
#+title: My Doom Emacs Config
#+author: wzj
#+property: header-args :tangle config.el

* 个人信息
#+begin_src emacs-lisp
(setq user-full-name "wzj"
      user-mail-address "wzj19952545705@126.com")
#+end_src
* 字体
#+begin_src emacs-lisp
(setq doom-font (font-spec :family "FiraCode Nerd Font" :size 20)
      doom-variable-pitch-font (font-spec :family "WenQuanYi Zen Hei" :size 18)
      doom-serif-font (font-spec :family "DejavuSans Mono" :size 19)
      doom-unicode-font (font-spec :family "FontAwesome" :size 19))
#+end_src
* 主题
#+begin_src emacs-lisp
(setq doom-theme 'doom-one)
#+end_src
* 相对+绝对行号
#+begin_src emacs-lisp
(setq display-line-numbers-type 'relative
      global-display-line-numbers-mode t)
#+end_src
* Org mode 个人化
** org agenda文件
#+begin_src emacs-lisp
(setq org-agenda-files '("~/nc/Org/agenda.org"))
#+end_src
** 实时载入 .org 调用代码到同名 .el 文件
#+begin_src emacs-lisp
(use-package! org-auto-tangle
  :defer t
  :hook (org-mode . org-auto-tangle-mode)
  :config (setq org-auto-tangle-default t))
#+end_src
** Xelatex 编译链
#+begin_src emacs-lisp
(setq org-latex-pdf-process '("xelatex -interaction nonstopmode %f"
                              "xelatex -interaction nonstopmode %f"))
#+end_src
** 采用 org-modern 样式
*** 安装 org-modern
#+begin_src emacs-lisp
(use-package! org-modern)
(global-org-modern-mode 1)
#+end_src
*** Github官方建议的一些搭配设置
#+begin_src emacs-lisp
(modify-all-frames-parameters
 '((right-divider-width . 40)
   (internal-border-width . 40)))
(dolist (face '(window-divider
                window-divider-first-pixel
               window-divider-last-pixel))
 (face-spec-reset-face face)
 (set-face-foreground face (face-attribute 'default :background)))
(set-face-background 'fringe (face-attribute 'default :background))

(setq
; Edit settings
 org-auto-align-tags nil
 org-tags-column 0
 org-fold-catch-invisible-edits 'show-and-error
 org-special-ctrl-a/e t
 org-insert-heading-respect-content t
; Org styling, hide markup etc.
 org-hide-emphasis-markers t
 org-pretty-entities t
org-ellipsis " ▼ "
; Agenda styling
org-agenda-tags-column 0
org-agenda-block-separator ?─
org-agenda-time-grid
'((daily today require-timed)
  (800 1000 1200 1400 1600 1800 2000)
  " ┄┄┄┄┄ " "┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄")
org-agenda-current-time-string
"⭠ now ─────────────────────────────────────────────────")
#+end_src
* 终端 Vterm
** 我用 fish
#+begin_src emacs-lisp
(setq shell-file-name "/usr/bin/fish"
      vterm-max-scrollback 5000)
#+end_src


.emacs.d/config.org
#+TITLE: 我的Emacs配置
#+AUTHOR: wzj

* 基础设置
** 垃圾回收
#+begin_src emacs-lisp
  (setq gc-cons-threshold (* 50 1000 1000))
#+end_src
** 去除无用工具栏
#+begin_src emacs-lisp
  (tool-bar-mode -1)
  (scroll-bar-mode -1)
  (menu-bar-mode -1)
#+end_src
** 设置字体
#+begin_src emacs-lisp
  (set-face-attribute 'default nil :font "Fira Code Retina" :height 180)
  (set-face-attribute 'fixed-pitch nil :font "Fira Code Retina" :height 180)
  (set-face-attribute 'variable-pitch nil :font "DejavuSans Mono" :height 180 :weight 'regular)
#+end_src
** 设置行列号
#+begin_src emacs-lisp
  (global-display-line-numbers-mode t)
#+end_src
** 设置主题
#+begin_src emacs-lisp
(use-package doom-themes
  :init (load-theme 'doom-palenight t))
#+end_src
** 设置状态栏
#+begin_src emacs-lisp
  (use-package all-the-icons)
  (use-package doom-modeline
    :init (doom-modeline-mode 1)
    :custom ((doom-modeline-height 15)))
#+end_src
** 初始面板
#+begin_src emacs-lisp
  (use-package dashboard
    :ensure t
    :custom (setq dashboard-center-content t)
    :config (dashboard-setup-startup-hook))
#+end_src
** 丝滑滚动
#+begin_src emacs-lisp
 (setq scroll-step 1
       scroll-conservatively 10000
       auto-window-vscroll nil)
#+end_src
** 彩虹括号
#+begin_src emacs-lisp
(use-package rainbow-delimiters
  :hook (prog-mode . rainbow-delimiters-mode))
#+end_src
* 插件源自动更新
#+begin_src emacs-lisp
 (use-package auto-package-update
  :custom
  (auto-package-update-interval 7)
  (auto-package-update-prompt-before-update t)
  (auto-package-update-hide-results t)
  :config
  (auto-package-update-maybe)
  (auto-package-update-at-time "09:00"))
#+end_src
* Org mode 优化
** Org mode 基础配置
#+begin_src emacs-lisp
(modify-all-frames-parameters
 '((right-divider-width . 40)
   (internal-border-width . 40)))
(dolist (face '(window-divider
                window-divider-first-pixel
                window-divider-last-pixel))
  (face-spec-reset-face face)
  (set-face-foreground face (face-attribute 'default :background)))
(set-face-background 'fringe (face-attribute 'default :background))

(setq
 org-auto-align-tags nil
 org-tags-column 0
 org-catch-invisible-edits 'show-and-error
 org-special-ctrl-a/e t
 org-insert-heading-respect-content t

 org-hide-emphasis-markers t
 org-pretty-entities t
 org-ellipsis "…"

 org-agenda-tags-column 0
 org-agenda-block-separator ?─
 org-agenda-time-grid
 '((daily today require-timed)
   (800 1000 1200 1400 1600 1800 2000)
   " ┄┄┄┄┄ " "┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄")
 org-agenda-current-time-string
 "⭠ now ─────────────────────────────────────────────────")
#+end_src
** 使用Org-modern
#+begin_src emacs-lisp
(use-package org-modern
  :init (global-org-modern-mode))
#+end_src

* 仿 Vim 快捷键
#+begin_src emacs-lisp
;; Make ESC quit prompts
(global-set-key (kbd "<escape>") 'keyboard-escape-quit)

(use-package general
  :after evil
  :config
  (general-create-definer wzj/leader-keys
    :keymaps '(normal insert visual emacs)
    :prefix "SPC"
    :global-prefix "C-SPC")

  (wzj/leader-keys
    "t"  '(:ignore t :which-key "toggles")
    "tt" '(counsel-load-theme :which-key "choose theme")
    "fde" '(lambda () (interactive) (find-file (expand-file-name "~/.emacs.d/Emacs.org")))))

(use-package evil
  :init
  (setq evil-want-integration t)
  (setq evil-want-keybinding nil)
  (setq evil-want-C-u-scroll t)
  (setq evil-want-C-i-jump nil)
  :config
  (evil-mode 1)
  (define-key evil-insert-state-map (kbd "C-g") 'evil-normal-state)
  (define-key evil-insert-state-map (kbd "C-h") 'evil-delete-backward-char-and-join)

  ;; Use visual line motions even outside of visual-line-mode buffers
  (evil-global-set-key 'motion "j" 'evil-next-visual-line)
  (evil-global-set-key 'motion "k" 'evil-previous-visual-line)

  (evil-set-initial-state 'messages-buffer-mode 'normal)
  (evil-set-initial-state 'dashboard-mode 'normal))

(use-package evil-collection
  :after evil
  :config
  (evil-collection-init))
#+end_src
* Ivy自动补全
#+begin_src emacs-lisp
  (use-package ivy
    :diminish
    :bind (("C-s" . swiper)
	   :map ivy-minibuffer-map
	   ("TAB" . ivy-alt-done)
	   ("C-l" . ivy-alt-done)
	   ("C-j" . ivy-next-line)
	   ("C-k" . ivy-previous-line)
	   :map ivy-switch-buffer-map
	   ("C-k" . ivy-previous-line)
	   ("C-l" . ivy-done)
	   ("C-d" . ivy-switch-buffer-kill)
	   :map ivy-reverse-i-search-map
	   ("C-k" . ivy-previous-line)
	   ("C-d" . ivy-reverse-i-search-kill))
    :config
    (ivy-mode 1))

  (use-package ivy-rich
    :after ivy
    :init
    (ivy-rich-mode 1))

  (use-package counsel
    :bind (("C-M-j" . 'counsel-switch-buffer)
	   :map minibuffer-local-map
	   ("C-r" . 'counsel-minibuffer-history))
    :custom
    (counsel-linux-app-format-function #'counsel-linux-app-format-function-name-only)
    :config
    (counsel-mode 1))

(use-package ivy-prescient
  :after counsel
  :custom
  (ivy-prescient-enable-filtering nil)
  :config
  ;; Uncomment the following line to have sorting remembered across sessions!
  ;(prescient-persist-mode 1)
  (ivy-prescient-mode 1))
#+end_src


.doom.d/config.el
(setq user-full-name "wzj"
      user-mail-address "wzj19952545705@126.com")

(setq doom-font (font-spec :family "FiraCode Nerd Font" :size 20)
      doom-variable-pitch-font (font-spec :family "WenQuanYi Zen Hei" :size 18)
      doom-serif-font (font-spec :family "DejavuSans Mono" :size 19)
      doom-unicode-font (font-spec :family "FontAwesome" :size 19))

(setq doom-theme 'doom-one)

(setq display-line-numbers-type 'relative
      global-display-line-numbers-mode t)

(setq org-agenda-files '("~/nc/Org/agenda.org"))

(use-package! org-auto-tangle
  :defer t
  :hook (org-mode . org-auto-tangle-mode)
  :config (setq org-auto-tangle-default t))

(setq org-latex-pdf-process '("xelatex -interaction nonstopmode %f"
                              "xelatex -interaction nonstopmode %f"))

(use-package! org-modern)
(global-org-modern-mode 1)

(modify-all-frames-parameters
 '((right-divider-width . 40)
   (internal-border-width . 40)))
(dolist (face '(window-divider
                window-divider-first-pixel
               window-divider-last-pixel))
 (face-spec-reset-face face)
 (set-face-foreground face (face-attribute 'default :background)))
(set-face-background 'fringe (face-attribute 'default :background))

(setq
; Edit settings
 org-auto-align-tags nil
 org-tags-column 0
 org-fold-catch-invisible-edits 'show-and-error
 org-special-ctrl-a/e t
 org-insert-heading-respect-content t
; Org styling, hide markup etc.
 org-hide-emphasis-markers t
 org-pretty-entities t
org-ellipsis " ▼ "
; Agenda styling
org-agenda-tags-column 0
org-agenda-block-separator ?─
org-agenda-time-grid
'((daily today require-timed)
  (800 1000 1200 1400 1600 1800 2000)
  " ┄┄┄┄┄ " "┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄")
org-agenda-current-time-string
"⭠ now ─────────────────────────────────────────────────")

(setq shell-file-name "/usr/bin/fish"
      vterm-max-scrollback 5000)
